\nonstopmode{}
\documentclass[a4paper]{book}
\usepackage[times,inconsolata,hyper]{Rd}
\usepackage{makeidx}
\usepackage[utf8,latin1]{inputenc}
% \usepackage{graphicx} % @USE GRAPHICX@
\makeindex{}
\begin{document}
\chapter*{}
\begin{center}
{\textbf{\huge Package `Rphylip'}}
\par\bigskip{\large \today}
\end{center}
\begin{description}
\raggedright{}
\item[Version]\AsIs{0.1-13}
\item[Date]\AsIs{2013-12-21}
\item[Title]\AsIs{Rphylip: An R interface for PHYLIP}
\item[Author]\AsIs{Liam J. Revell}
\item[Maintainer]\AsIs{Liam J. Revell }\email{liam.revell@umb.edu}\AsIs{}
\item[Depends]\AsIs{R (>= 2.10), ape (>= 3.0-10)}
\item[ZipData]\AsIs{no}
\item[Description]\AsIs{Rphylip provides an R interface for the PHYLIP package. All users of
Rphylip will thus first have to install the PHYLIP phylogeny methods program
package (Felsenstein 2013).
See http://evolution.genetics.washington.edu/phylip.html for more information
about installing PHYLIP.}
\item[License]\AsIs{GPL (>= 2)}
\item[URL]\AsIs{}\url{http://www.phytools.org/Rphylip}\AsIs{}
\item[Repository]\AsIs{}
\item[Date/Publication]\AsIs{2013-12-21 12:00:00 EDT}
\end{description}
\Rdcontents{\R{} topics documented:}
\inputencoding{utf8}
\HeaderA{Rphylip-package}{Rphylip: An R interface for PHYLIP}{Rphylip.Rdash.package}
\aliasA{Rphylip}{Rphylip-package}{Rphylip}
\keyword{package}{Rphylip-package}
%
\begin{Description}\relax
\pkg{Rphylip} provides an R interface for programs in the PHYLIP phylogeny methods package (Felsenstein 1989, 2013).
\end{Description}
%
\begin{Details}\relax
The complete list of functions can be displayed with \code{library(help = Rphylip)}.

Obviously, before any of the functions of this package can be used, users must first install PHYLIP (Felsenstein 2013). More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.

More information on \pkg{Rphylip} can be found at \url{http://www.phytools.org/Rphylip/} or \url{http://blog.phytools.org}. The latest code for the development version of Rphylip can also be found on github at the following URL: \url{http://github.com/liamrevell/Rphylip}.
\end{Details}
%
\begin{Author}\relax
Liam J. Revell

Maintainer: Liam J. Revell <liam.revell@umb.edu>
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Revell, L. J. (2013) Rphylip: An R interface for PHYLIP. R package version 0-1.09.
\end{References}
\inputencoding{utf8}
\HeaderA{as.proseq}{Converts objects to protein sequences}{as.proseq}
\keyword{utilities}{as.proseq}
\keyword{amino acid}{as.proseq}
%
\begin{Description}\relax
Converts objects to class \code{"proseq"}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
as.proseq(x, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{x}] an object containing amino sequences. (Presently only objects of class \code{"phyDat"} are permitted.)
\item[\code{...}] optional arguments.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
An object of class \code{"proseq"} containing protein sequences.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{SeeAlso}\relax
\code{\LinkA{print.proseq}{print.proseq}}, \code{\LinkA{Rproml}{Rproml}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{opt.Rdnaml}{Parameter optimizer for Rdnaml}{opt.Rdnaml}
\keyword{phylogenetics}{opt.Rdnaml}
\keyword{inference}{opt.Rdnaml}
\keyword{maximum likelihood}{opt.Rdnaml}
\keyword{DNA}{opt.Rdnaml}
%
\begin{Description}\relax
This function is an wrapper for \code{\LinkA{Rdnaml}{Rdnaml}} that attempts to optimize \code{gamma} (the alpha shape parameter of the gamma model of rate heterogeneity among sites), \code{kappa} (the transition:transversion ratio), and \code{bf} (the base frequencies).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
opt.Rdnaml(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing dnaml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{tree} fixed tree to use in optimization - if not provided, it will be estimated using \code{\LinkA{Rdnaml}{Rdnaml}} under the default conditions; \code{bounds} a list with bounds for optimization - for \code{kappa} and \code{gamma} this should be a two-element vector, whereas for \code{bf} this should be a 4 x 2 matrix with lower bounds in column 1 and upper bounds in column 2.

More information about the dnaml program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/dnaml.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns a list with the following components: \code{kappa}, \code{gamma}, \code{bf} (see Details), and \code{logLik} (the log-likelihood of the fitted model).
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (1981) Evolutionary trees from DNA sequences: A Maximum Likelihood approach. \emph{Journal of Molecular Evolution}, 17, 368-376.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Felsenstein, J., Churchill, G. A. (1996) A Hidden Markov Model approach to variation among sites in rate of evolution. \emph{Molecular Biology and Evolution}, 13, 93-104.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnaml}{Rdnaml}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
fit<-opt.Rdnaml(primates,bounds=list(kappa=c(0.1,40))
tree<-Rdnaml(primates,kappa=fit$kappa,gamma=fit$gamma,bf=fit$bf)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{primates}{Example datasets}{primates}
\aliasA{chloroplast}{primates}{chloroplast}
\keyword{datasets}{primates}
%
\begin{Description}\relax
\code{primates} is an object of class \code{"DNAbin"} containing nucleotide sequence data of mysterious origin for 12 species of primates. \code{chloroplast} is an object of class\code{"proseq"} containing a chloroplast alignment from the phangorn package (Schliep 2011).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(primates)
data(chloroplast)
\end{verbatim}
\end{Usage}
%
\begin{Format}
The data are stored as an object of class \code{"DNAbin"} or \code{"proseq"}.
\end{Format}
%
\begin{Source}\relax
Unknown.
\end{Source}
\inputencoding{utf8}
\HeaderA{print.proseq}{Print method protein sequences}{print.proseq}
\keyword{utilities}{print.proseq}
\keyword{amino acid}{print.proseq}
%
\begin{Description}\relax
Print method for an object of class \code{"proseq"}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
## S3 method for class 'proseq'
print(x, printlen=6, digits=3, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{x}] an object of class \code{"proseq"}.
\item[\code{printlen}] number of sequence names to print.
\item[\code{digits}] number of digits to print.
\item[\code{...}] optional arguments.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Prints to screen.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{SeeAlso}\relax
\code{\LinkA{as.proseq}{as.proseq}}, \code{\LinkA{print.DNAbin}{print.DNAbin}}, \code{\LinkA{Rproml}{Rproml}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rconsense}{R interface for consense}{Rconsense}
\keyword{phylogenetics}{Rconsense}
\keyword{consensus}{Rconsense}
%
\begin{Description}\relax
This function is an R interface for consense in the PHYLIP package (Felsenstein 2013). consense can be used to compute the consensus tree from a set of phylogenies. 
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rconsense(trees, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{trees}] an object of class \code{"multiPhylo"}.
\item[\code{path}] path to the directory containing the executable consense. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to consense. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{method} which can be \code{"extended"} (extended majority rule consensus, the default), \code{"strict"} (strict consensus), or regular majority rule consensus (\code{"majority"}); \code{"outgroup"} single taxon label or vector of taxa that should be used to root all trees before analysis; \code{rooted} logical value indicated whether to treat the input trees as rooted (defaults to \code{rooted = FALSE}); and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the consense program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/consense.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"}. For methods other than \code{method = "strict"}, \code{tree\$node.label} contains the proportion of phylogenies in \code{trees} containing that subtree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Margush, T., McMorris, F.R. (1981) Consensus n-trees. \emph{Bulletin of Mathematical Biology}, 43, 239-244.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
trees<-rmtree(n=10,N=10)
tree<-Rconsense(trees)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rcontml}{R interface for contml}{Rcontml}
\keyword{phylogenetics}{Rcontml}
\keyword{inference}{Rcontml}
\keyword{maximum likelihood}{Rcontml}
\keyword{continuous characters}{Rcontml}
%
\begin{Description}\relax
This function is an R interface for contml in the PHYLIP package (Felsenstein 2013). contml can be used for ML phylogeny estimation from gene frequency data or continuous characters. The continuous characters should be rotated so as to be uncorrelated before analysis.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rcontml(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] either (a) a \emph{matrix} of continuous valued traits (in columns) with rownames containing species names; or (b) a list of matrices in which each row contains the relative frequency of alleles at a locus for a species. In the latter case the rownames of each matrix in the list should contain the species names.
\item[\code{path}] path to the executable containing contml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to contml. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the model will be optimized on a fixed input topology; \code{global} perform global search (defaults to \code{global = TRUE}); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input/output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the contml program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/contml.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (1981) Maximum likelihood estimation of evolutionary trees from continuous characters. \emph{American Journal of Human Genetics}, 25, 471-492.

Felsenstein, J. (1981) Maximum likelihood estimation of evolutionary trees from continuous characters. \emph{American Journal of Human Genetics}, 25, 471-492.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnaml}{Rdnaml}}, \code{\LinkA{Rproml}{Rproml}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rcontrast}{R interface for contrast}{Rcontrast}
\keyword{phylogenetics}{Rcontrast}
\keyword{comparative method}{Rcontrast}
\keyword{continuous characters}{Rcontrast}
\keyword{maximum likelihood}{Rcontrast}
%
\begin{Description}\relax
This function is an R interface for contrast in the PHYLIP package (Felsenstein 2013). contrast can be used to perform the among species phylogenetically independent contrasts method of Felsenstein (1985) and the within \& among species method of Felsenstein (2008).

More information about the contrast program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/contrast.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rcontrast(tree, X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{tree}] object of class \code{"phylo"}.
\item[\code{X}] a \emph{matrix} of continuous valued traits (in columns) with rownames containing species names. For one trait, \code{X} can be a matrix with one column or a vector with \code{names(X)} containing species names matching \code{tree\$tip.label}. For within-species contrasts analysis, the matrix should contain repeating (identical) row names for conspecifics.
\item[\code{path}] path to the executable containing contrast. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to contrast. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); and \code{cleanup} remove PHYLIP input/output files after the analysis is completed (defaults to \code{cleanup = TRUE}).
\end{Details}
%
\begin{Value}
If \code{X} contains one observation per species (say, the species mean), then \code{Rcontrast} returns a list with the following components: \code{Contrasts}, a matrix with all phylogenetically independent contrats; \code{Covariance\_matrix}, a matrix containing the evolutionary variances (on diagonals) and covariances; \code{Regressions}, a matrix containing the pair-wise bivariate regression coefficients (columns on rows); \code{Correlations}, a correlation matrix of contrasts.

If \code{X} contains more than one sample per species, then \code{Rcontrast} returns a list with the following elements: \code{VarA}, the estimated among-species variance-covariance matrix; \code{VarE}, the estimated within-species (i.e., 'environmental') variance-covariance matrix; \code{VarA.Regression}, a matrix containing the pair-wise bivariate among-species regression coefficients (columns on rows); \code{VarA.Correlations}, a matrix with the among-species evolutionary correlations; \code{VarE.Regressions}, the pair-wise bivariate within-species regression coefficients; \code{VarE.Correlations}, the within-species correlations; \code{nonVa.VarE}, \code{nonVa.VarE.Regressions}, and \code{nonVa.VarA.Correlations}, estimates obtained when \code{VarA} is not included in the model; \code{logLik} and \code{nonVa.logLik}, log-likelihood when \code{VarA} is included (or not) in the model; \code{k} and \code{nonVa.k} the number of parameters estimated in each model; and \code{P} the p-value of a likelihood-ratio test of \code{VarA}, in which \code{df = k - nonVa.k}. 
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (1985) Phylogenies and the comparative method. \emph{American Naturalist}, 125, 1-15.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2008) Comparative methods with sampling error and within-species variation: Contrasts revisited and revised. \emph{American Naturalist}, 171, 713-725.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{pic}{pic}}, \code{Rcontml}, \code{Rthreshml}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rdnacomp}{R interface for dnacomp}{Rdnacomp}
\keyword{phylogenetics}{Rdnacomp}
\keyword{inference}{Rdnacomp}
\keyword{parsimony}{Rdnacomp}
%
\begin{Description}\relax
This function is an R interface for dnacomp in the PHYLIP package (Felsenstein 2013). dnacomp can be used for phylogeny inference from DNA sequences using the compatibility method (Le Quesne 1969; Fitch 1975).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnacomp(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing dnacomp. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnacomp. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the number of compatible sites will be computed on a fixed input topology; \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the dnacomp program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/dnacomp.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Fitch, W.M. (1975) Toward finding the tree of maximum parsimony. pp. 189-230 in \emph{Proceedings of the Eighth International Conference on Numerical Taxonomy}, ed. G. F. Estabrook. W. H. Freeman, San Francisco.

Le Quesne, W.J. (1969) A method of selection of characters in numerical taxonomy. \emph{Systematic Zoology}, 18, 201-205. 
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnapars}{Rdnapars}}, \code{\LinkA{Rdnapenny}{Rdnapenny}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
tree<-Rdnacomp(primates)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rdnadist}{R interfaces for dnadist}{Rdnadist}
\keyword{phylogenetics}{Rdnadist}
\keyword{inference}{Rdnadist}
\keyword{maximum likelihood}{Rdnadist}
\keyword{distance method}{Rdnadist}
%
\begin{Description}\relax
This function is an R interface for dnadist in the PHYLIP package (Felsenstein 2013). dnadist can be used to estimate the evolutionary distances between DNA sequences under various models.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnadist(X, method=c("F84","K80","JC","LogDet"), path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{method}] method for calculating the distances. Can be \code{"F84"} (Kishino \& Hasegawa 1989; Felsenstein \& Churchill 1996), \code{"K80"} (Kimura 1980), \code{"JC"} (Jukes \& Cantor 1969), or \code{"LogDet"} (Barry \& Hartigan 1987; Lake 1994; Steel 1994; Lockhart et. al. 1994). Also \code{method="similarity"} computes the sequence similarity among the rows of \code{X}.
\item[\code{path}] path to the executable containing dnadist. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnadist. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{gamma} alpha shape parameter of a gamma model of rate heterogeneity among sites (defaults to no gamma rate heterogeneity); \code{kappa} transition:transversion ratio (defaults to \code{kappa = 2.0}); \code{rates} vector of rates (defaults to single rate); \code{rate.categories} vector of rate categories corresponding to the order of \code{rates}; \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{bf} vector of base frequencies in alphabetical order (i.e., A, C, G, \& T) - if not provided, then defaults to empirical frequencies; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the dnadist program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/dnadist.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"dist"}.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax

Barry, D., Hartigan, J.A. (1987) Statistical analysis of hominoid molecular evolution. \emph{Statistical Science}, 2, 191-200.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Felsenstein, J., Churchill, G. A. (1996) A Hidden Markov Model approach to variation among sites in rate of evolution. \emph{Molecular Biology and Evolution}, 13, 93-104.

Jukes, T.H., Cantor, C.R. (1969) Evolution of protein molecules. pp. 21-132 in \emph{Mammalian Protein Metabolism Vol. III}, ed. M.N. Munro. Academic Press, New York.

Kimura, M. (1980) A simple model for estimating evolutionary rates of base substitutions through comparative studies of nucleotide sequences. \emph{Journal of Molecular Evolution}, 16, 111-120.

Kishino, H., Hasegawa, M. (1989) Evaluation of the maximum likelihood estimate of teh evolutionary tree topology from DNA sequence data, and the branching order in Hominoidea. \emph{Journal of Molecular Evolutioon}, 29, 170-179.

Lake, J.A. (1994) Reconstructing evolutionary trees from DNA and protein sequences: Paralinear distances. \emph{Proceedings of the National Academy of Sciences}, 91, 1455-1459.

Lockhart, P.J., Steel, M.A., Hendy, M.D., Penny, D. (1994) Recovering evolutionary trees under a more realistic model of sequence evolution. \emph{Molecular Biology and Evolution}, 11, 605-612.

Steel, M.A. (1994) Recovering a tree from the Markov leaf colourations it generates under a Markov model. \emph{Applied Mathematics Letters}, 7, 19-23.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rneighbor}{Rneighbor}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
D<-Rdnadist(primates,kappa=10)
tree<-Rneighbor(D)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rdnainvar}{R interface for dnainvar}{Rdnainvar}
\keyword{phylogenetics}{Rdnainvar}
\keyword{inference}{Rdnainvar}
%
\begin{Description}\relax
This function is an R interface for dnainvar in the PHYLIP package (Felsenstein 2013). dnainvar reads DNA sequences and computest Lake's invariants (Cavender \& Felsenstein 1987; Lake 1987).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnainvar(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}. \code{X} cannot contain more than four species.
\item[\code{path}] path to the executable containing dnainvar. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnainvar. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the dnainvar program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/dnainvar.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
At present, this function merely prints the results of dnainvar to screen.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Cavender, J.A., Felsenstein, J. (1987) Invariants of phylogenies in a simple case with discrete states. \emph{Journal of Classification}, 4, 57-71. 

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Lake, J.A. (1987) A rate-independent technique for analysis of nucleic acid sequences: evolutionary parsimony. \emph{Molecular Biology and Evolution}, 4, 167-191. 
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnapars}{Rdnapars}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
primates<-primates[sample(nrow(primates),4),]
tree<-Rdnainvar(primates)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rdnaml}{R interfaces for dnaml and dnamlk}{Rdnaml}
\aliasA{Rdnamlk}{Rdnaml}{Rdnamlk}
\keyword{phylogenetics}{Rdnaml}
\keyword{inference}{Rdnaml}
\keyword{maximum likelihood}{Rdnaml}
%
\begin{Description}\relax
This function is an R interface for dnaml in the PHYLIP package (Felsenstein 2013). dnaml can be used for ML phylogeny estimation from DNA sequences (Felsenstein 1981; Felsenstein \& Churchill 1996).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnaml(X, path=NULL, ...)
Rdnamlk(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing dnaml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnaml or dnamlk. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the model will be optimized on a fixed input topology; \code{kappa} transition:transversion ratio (defaults to \code{kappa = 2.0}); \code{bf} vector of base frequencies in alphabetical order (i.e., A, C, G, \& T) - if not provided, then defaults to empirical frequencies; \code{rates} vector of rates (defaults to single rate); \code{rate.categories} vector of rate categories corresponding to the order of \code{rates}; \code{gamma} alpha shape parameter of a gamma model of rate heterogeneity among sites (defaults to no gamma rate heterogeneity); \code{ncat} number of rate categories for the gamma model; \code{inv} proportion of invariant sites for the invariant sites model (defaults to \code{inv = 0}); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{speedier} speedier but rougher analysis (defaults to \code{speedier = FALSE}); \code{global} perform global search (defaults to \code{global = TRUE}); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Finally \code{clock=TRUE} enforces a molecular clock. The argument \code{clock} is only available for \code{Rdnaml}. If \code{clock=TRUE} then dnamlk is used internally. For \code{Rdnamlk} a molecular clock is assumed, thus \code{Rdnaml(...,clock=TRUE)} and \code{Rdnamlk(...)} are equivalent.

More information about the dnaml and dnamlk programs in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/dnaml.html}, and here \url{http://evolution.genetics.washington.edu/phylip/doc/dnamlk.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (1981) Evolutionary trees from DNA sequences: A Maximum Likelihood approach. \emph{Journal of Molecular Evolution}, 17, 368-376.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Felsenstein, J., Churchill, G. A. (1996) A Hidden Markov Model approach to variation among sites in rate of evolution. \emph{Molecular Biology and Evolution}, 13, 93-104.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{opt.Rdnaml}{opt.Rdnaml}}, \code{\LinkA{Rcontml}{Rcontml}}, \code{\LinkA{Rproml}{Rproml}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
tree<-Rdnaml(primates,kappa=10)
clockTree<-Rdnamlk(primates,kappa=10)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rdnapars}{R interface for dnapars}{Rdnapars}
\keyword{phylogenetics}{Rdnapars}
\keyword{inference}{Rdnapars}
\keyword{parsimony}{Rdnapars}
%
\begin{Description}\relax
This function is an R interface for dnapars in the PHYLIP package (Felsenstein 2013). dnapars can be used for MP phylogeny estimation from DNA sequences (Eck \& Dayhoff 1966; Kluge \& Farris 1969; Fitch 1971).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnapars(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing dnapars. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnapars. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the parsimony score will be computed on a fixed input topology; \code{thorough} logical value indicating whether to conduct a more thorough search (defaults to \code{thorough=TRUE}); \code{nsave} number of trees to save if multiple equally parsimonious trees are found (defaults to \code{nsave=10000}); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{threshold} threshold value for threshold parsimony (defaults to ordinary parsimony); \code{transversion} logical value indicating whether to use transversion parsimony (defaults to \code{transversion=FALSE}); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the dnapars program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/dnapars.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Eck. R.V., Dayhoff, M.O. (1966) \emph{Atlas of Protein Sequence and Structure 1966}. National Biomedical Research Foundation, Silver Spring, Maryland.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Fitch, W.M. (1971) Toward defining the course of evolution: Minimu change for a specified tree topology. \emph{Systematic Zoology}, 20, 406-416.

Kluge, A.G., Farris, J.S. (1969) Quantitative phyletics and the evolution of anurans. \emph{Systematic Zoology}, 18, 1-32.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnaml}{Rdnaml}}, \code{\LinkA{Rdnapenny}{Rdnapenny}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
tree<-Rdnapars(primates)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rdnapenny}{R interface for dnapenny}{Rdnapenny}
\keyword{phylogenetics}{Rdnapenny}
\keyword{inference}{Rdnapenny}
\keyword{parsimony}{Rdnapenny}
%
\begin{Description}\relax
This function is an R interface for dnapenny in the PHYLIP package (Felsenstein 2013). dnapenny performs branch \& bound parsimony searching following Hendy \& Penny (1982).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnapenny(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing dnapenny. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnapenny. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{groups} number of groups of 1,000 trees (defaults to \code{groups = 10000}); \code{report} reporting frequency, in numbers of trees (defaults to \code{report = 1000}); \code{simple} simple branch \& bound (defaults to \code{simple = TRUE}); \code{threshold} threshold value for threshold parsimony (defaults to ordinary parsimony); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the dnapenny program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/dnapenny.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} or \code{"multiPhylo"} that is the tree or trees with the best parsimony score. \code{tree\$score} gives the parsimony score, for \code{"phylo"} object \code{tree}.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Hendy, M.D., Penny, D. (1982) Branch and bound algorithms to determine minimal evolutionary trees. \emph{Mathematical Biosciences}, 60, 133-142.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnapars}{Rdnapars}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
tree<-Rdnapenny(primates)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{read.protein}{Reads protein sequences from file in multiple formats}{read.protein}
\keyword{phylogenetics}{read.protein}
\keyword{utilities}{read.protein}
\keyword{amino acid}{read.protein}
%
\begin{Description}\relax
Reads protein sequences from a file.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
read.protein(file, format="fasta", ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{file}] file name for file containing protein sequences.
\item[\code{format}] format of input file. Permitted formats are \code{"fasta"} and \code{"sequential"}. See \code{\LinkA{read.dna}{read.dna}} for more information.
\item[\code{...}] optional arguments.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
An object of class \code{"proseq"} containing protein sequences.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{SeeAlso}\relax
\code{\LinkA{as.proseq}{as.proseq}}, \code{\LinkA{print.proseq}{print.proseq}}, \code{\LinkA{Rproml}{Rproml}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rfitch}{R interface for fitch}{Rfitch}
\keyword{phylogenetics}{Rfitch}
\keyword{inference}{Rfitch}
\keyword{distance method}{Rfitch}
%
\begin{Description}\relax
This function is an R interface for fitch in the PHYLIP package (Felsenstein 2013). fitch can be used for tree inference using the Fitch-Margoliash method (Fitch \& Margoliash 1967), the Cavalli-Sforza \& Edwards leasts-squares method (Cavalli-Sforza \& Edwards 1967), and the minimum evolution criterion (Kidd \& Sgaramella-Zonta 1971; Nei \& Rzhetsky 1993).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rfitch(D, path=NULL , ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{D}] a distance matrix as an object of class \code{"matrix"} or \code{"dist"}. If a matrix, then \code{D} should be symmetrical with a diagonal of zeros.
\item[\code{path}] path to the executable containing fitch. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to fitch. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{method} - can be \code{"FM"} or \code{"fm"} (for the Fitch-Margoliash criterion; Fitch \& Margoliash 1967), \code{"LS"} or \code{"ls"} for the Cavalli-Sforza \& Edwards (1967) least-squares criterion, or \code{"ME"} or \code{"me"} (for Minimum Evolution); \code{tree} object of class \code{"phylo"} - if supplied, then branch lengths are optimized \& the optimality criterion is computed on the input tree; \code{negative} a logical value indicating whether negative branch lengths should be permitted (defaults to \code{negative = TRUE}; \code{global} perform global search (defaults to \code{global = TRUE}); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the fitch program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/fitch.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the NJ or UPGMA tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Cavalli-Sforza, L.L., Edwards, A.W.F. (1967) Phylogenetic analysis: models and estimation procedures. \emph{Evolution}, 32, 550-570. 

Fitch, W.M., Margoliash, E. (1967) Construction of phylogenetic trees. \emph{Science}, 155, 279-284.

Kidd, K.K., Sgaramella-Zonta, L.A. (1971) Phylogenetic analysis: concepts and methods. \emph{American Journal of Human Genetics}, 23, 235-252. 	

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnadist}{Rdnadist}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
D<-dist.dna(data(primates),model="JC")
tree<-Rfitch(D)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rneighbor}{R interface for neighbor}{Rneighbor}
\keyword{phylogenetics}{Rneighbor}
\keyword{inference}{Rneighbor}
\keyword{distance method}{Rneighbor}
%
\begin{Description}\relax
This function is an R interface for neighbor in the PHYLIP package (Felsenstein 2013). neighbor can be used for neighbor-joining (Saitou \& Nei 1987) and UPGMA (Sokal \& Michener 1958) phylogeny inference.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rneighbor(D, path=NULL , ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{D}] a distance matrix as an object of class \code{"matrix"} or \code{"dist"}. If a matrix, then \code{D} should be symmetrical with a diagonal of zeros.
\item[\code{path}] path to the executable containing neighbor. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to neighbor. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{method} - can be \code{"NJ"} or \code{"nj"} (for neighbor-joining), or \code{"UPGMA"} or \code{"UPGMA"} (for UPGMA); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired (only works for \code{method = "NJ"}, UPGMA trees are already rooted); and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the neighbor program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/neighbor.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the NJ or UPGMA tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Saitou, N., Nei, M. (1987) The neighbor-joining method: A new method for reconstructing phylogenetic trees. \emph{Molecular Biology and Evolution}, 4, 406-425.

Sokal, R., Michener, C. (1958) A statistical method for evaluating systematic relationships. \emph{University of Kansas Science Bulletin}, 38, 1409-1438.	

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnadist}{Rdnadist}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
D<-dist.dna(data(primates),model="JC")
tree<-Rneighbor(D)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rproml}{R interfaces for proml and promlk}{Rproml}
\aliasA{Rpromlk}{Rproml}{Rpromlk}
\keyword{phylogenetics}{Rproml}
\keyword{inference}{Rproml}
\keyword{maximum likelihood}{Rproml}
\keyword{amino acid}{Rproml}
%
\begin{Description}\relax
This function is an R interface for proml in the PHYLIP package (Felsenstein 1989, 2013). proml can be used for ML phylogeny estimation from amino acid sequences.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rproml(X, path=NULL, ...)
Rpromlk(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"proseq"}.
\item[\code{path}] path to the executable containing proml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to proml or promlk. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the model will be optimized on a fixed input topology; \code{model} amino acid model - could be \code{"JTT"} (Jones et al. 1992), \code{"PMB"} (Veerassamy et al. 2003), or \code{"PAM"} (Dayhoff \& Eck 1968; Dayhoff et al. 1979; Koisol \& Goldman 2005); \code{rates} vector of rates (defaults to single rate); \code{rate.categories} vector of rate categories corresponding to the order of \code{rates}; \code{gamma} alpha shape parameter of a gamma model of rate heterogeneity among sites (defaults to no gamma rate heterogeneity); \code{ncat} number of rate categories for the gamma model; \code{inv} proportion of invariant sites for the invariant sites model (defaults to \code{inv = 0}); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{speedier} speedier but rougher analysis (defaults to \code{speedier = FALSE}); \code{global} perform global search (defaults to \code{global = TRUE}); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Finally \code{clock=TRUE} enforces a molecular clock. The argument \code{clock} is only available for \code{Rproml}. If \code{clock=TRUE} then promlk is used internally. For \code{Rpromlk} a molecular clock is assumed, thus \code{Rproml(...,clock=TRUE)} and \code{Rpromlk(...)} are equivalent. Note that in PHYLIP 3.695 my tests of promlk yielded peculiar results (all branch lengths zero length, random topology), so I'm not sure what to make of that.

More information about the proml and promlk programs in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/proml.html}, and here \url{http://evolution.genetics.washington.edu/phylip/doc/promlk.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Dayhoff, M.O., Eck, R.V. (1968) \emph{Atlas of Protein Sequence and Structure 1967-1968}. National Biomedical Research Foundation, Silver Spring, Maryland.

Dayhoff, M.O., Schwartz, R.M., Orcutt, B.C. (1979) A model of evolutionary change in proteins. pp. 345-352 in \emph{Atlas of Protein Sequence and Structure, Volume 5, Supplement 3, 1978}, ed. M.O. Dayhoff. National Biomedical Research Foundataion, Silver Spring, Maryland.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Jones, D.T., Taylor, W.R., Thornton, J.M. (1992) The rapid generation of mutation data matrices from protein sequences. \emph{Computer Applications in the Biosciences (CABIOS)}, 8, 275-282.

Koisol, C., Goldman, N. (2005) Different versions of the Dayhoff rate matrix. \emph{Molecular Biology and Evolution}, 22, 193-199.

Veerassamy, S., Smith, A., Tillier, E.R. (2003) A transition probability model for amino acid substitutions from blocks. \emph{Journal of Computational Biology}, 10, 997-1010.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{as.proseq}{as.proseq}}, \code{\LinkA{Rdnaml}{Rdnaml}}, \code{\LinkA{read.protein}{read.protein}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(chloroplast)
tree<-Rproml(chloroplast)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rprotdist}{R interfaces for protdist}{Rprotdist}
\keyword{phylogenetics}{Rprotdist}
\keyword{maximum likelihood}{Rprotdist}
\keyword{distance method}{Rprotdist}
\keyword{amino acid}{Rprotdist}
%
\begin{Description}\relax
This function is an R interface for protdist in the PHYLIP package (Felsenstein 2013). protdist can be used to estimate the evolutionary distances between amino acid sequences under various models.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rprotdist(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"proseq"} containing aligned amino acid sequences.
\item[\code{path}] path to the executable containing protdist. If \code{path = NULL}, the R will search sev arguments to be passed to protdist. See details for more information.
\item[\code{...}] optional arguments to be passed to protdist. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{model} can be \code{"JTT"} (Jones et al. 1992), \code{"PMB"} (Veerassamy et al. 2003), \code{"PAM"} (Dayhoff \& Eck 1968; Dayhoff et al. 1979; Koisol \& Goldman 2005), \code{"Kimura"} (a simple model based on Kimura 1980), \code{"similarity"} which gives the similarity between sequences, and \code{"categories"} which is due to Felsenstein; \code{gamma} alpha shape parameter of a gamma model of rate heterogeneity among sites (defaults to no gamma rate heterogeneity) - note that gamma rate heterogeneity does not apply to \code{model = "Kimura"} or \code{model = "similarity"}; \code{kappa} transition:transversion ratio (defaults to \code{kappa = 2.0}), \code{genetic.code}, type of genetic code to assume (options are \code{"universal"}, the default, \code{"mitochondrial"}, \code{"vertebrate.mitochondrial"}, \code{"fly.mitochondrial"}, and \code{"yeast.mitochondrial"}), \code{categorization}, categorization scheme for amino acids (options are \code{"GHB"}, the George et al. 1988 classification, \code{"Hall"}, a classification scheme provided by Ben Hall, and \code{"chemical"}, a scheme based on Conn \& Stumpf 1963); and, finally, \code{ease}, a numerical parameter that indicates the facility of getting between amino acids of different categories in which 0 is nearly impossible, and 1 is no difficulty (defaults to \code{ease = 0.457}) - note that \code{kappa}, \code{bf}, \code{genetic.code}, \code{categorization}, and \code{ease} apply only to \code{model = "categories"}; \code{rates} vector of rates (defaults to single rate); \code{rate.categories} vector of rate categories corresponding to the order of \code{rates}; \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the protdist program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/protdist.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"dist"}.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Conn, E.E., Stumpf, P.K. (1963) \emph{Outlines of Biochemistry}. John Wiley and Sons, New York. 

Dayhoff, M.O., Eck, R.V. (1968) \emph{Atlas of Protein Sequence and Structure 1967-1968}. National Biomedical Research Foundation, Silver Spring, Maryland.

Dayhoff, M.O., Schwartz, R.M., Orcutt, B.C. (1979) A model of evolutionary change in proteins. pp. 345-352 in \emph{Atlas of Protein Sequence and Structure, Volume 5, Supplement 3, 1978}, ed. M.O. Dayhoff. National Biomedical Research Foundataion, Silver Spring, Maryland.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

George, D.G., Hunt, L.T., Barker., W.C. (1988) \emph{Current methods in sequence comparison and analysis}. pp. 127-149 in Macromolecular Sequencing and Synthesis, ed. D. H. Schlesinger. Alan R. Liss, New York. 

Jones, D.T., Taylor, W.R., Thornton, J.M. (1992) The rapid generation of mutation data matrices from protein sequences. \emph{Computer Applications in the Biosciences (CABIOS)}, 8, 275-282.

Kimura, M. (1980) A simple model for estimating evolutionary rates of base substitutions through comparative studies of nucleotide sequences. \emph{Journal of Molecular Evolution}, 16, 111-120.

Koisol, C., Goldman, N. (2005) Different versions of the Dayhoff rate matrix. \emph{Molecular Biology and Evolution}, 22, 193-199.

Veerassamy, S., Smith, A., Tillier, E.R. (2003) A transition probability model for amino acid substitutions from blocks. \emph{Journal of Computational Biology}, 10, 997-1010.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rneighbor}{Rneighbor}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(chloroplast)
D<-Rprotdist(chloroplast,model="PAM")
tree<-Rneighbor(D)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rprotpars}{R interface for protpars}{Rprotpars}
\keyword{phylogenetics}{Rprotpars}
\keyword{inference}{Rprotpars}
\keyword{parsimony}{Rprotpars}
\keyword{amino acid}{Rprotpars}
%
\begin{Description}\relax
This function is an R interface for protpars in the PHYLIP package (Felsenstein 2013). protpars can be used for MP phylogeny estimation from protein sequences (Eck \& Dayhoff 1966; Fitch 1971).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rprotpars(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"proseq"} containing aligned amino acid sequences.
\item[\code{path}] path to the executable containing protpars. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to protpars. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the parsimony score will be computed on a fixed input topology; \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{threshold} threshold value for threshold parsimony (defaults to ordinary parsimony); \code{genetic.code}, type of genetic code to assume (options are \code{"universal"}, the default, \code{"mitochondrial"}, \code{"vertebrate.mitochondrial"}, \code{"fly.mitochondrial"}, and \code{"yeast.mitochondrial"}); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the protpars program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/protpars.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Eck. R.V., Dayhoff, M.O. (1966) \emph{Atlas of Protein Sequence and Structure 1966}. National Biomedical Research Foundation, Silver Spring, Maryland.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Fitch, W.M. (1971) Toward defining the course of evolution: Minimu change for a specified tree topology. \emph{Systematic Zoology}, 20, 406-416.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{as.proseq}{as.proseq}}, \code{\LinkA{Rdnapars}{Rdnapars}}, \code{\LinkA{read.protein}{read.protein}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(chloroplast)
tree<-Rprotpars(chloroplast)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rthreshml}{R interface for threshml}{Rthreshml}
\keyword{phylogenetics}{Rthreshml}
\keyword{comparative method}{Rthreshml}
%
\begin{Description}\relax
This function is an R interface for threshml in the PHYLIP package (Felsenstein 1989, 2013). threshml fits the threshold model of Felsenstein (2005; 2012). Note that threshml is new \& not in the currently released version of PHYLIP (as of December 2013). It can be downloaded from its webpage here: \url{http://evolution.gs.washington.edu/phylip/download/threshml/}. If not specifying \code{path}, the executable file for threshml (e.g., threshml.exe in Windows) should be placed in the folder containing all other executable files for PHYLIP (e.g., C:/Program Files/phylip=3.695/exe in Windows).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rthreshml(tree, X, types=NULL, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{tree}] object of class \code{"phylo"}.
\item[\code{X}] a \emph{data.frame} of continuous valued or discrete character traits with rownames containing species names. Discrete \& continuous characters can be supplied in any order. All discrete character traits must be two-state, but can be coded using any convention (i.e., \code{0, 1}, \code{"A","B"}, etc.).
\item[\code{types}] character vector containing the types (e.g., \code{"discrete"}, \code{"continuous"}). If \code{types} are not supplied, \code{Rthreshml} will try to figure out which columns via \code{is.numeric}.
\item[\code{path}] path to the executable containing threshml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to threshml. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{burnin} burnin generations for the MCMC; \code{nchain} number of chains of the MCMC; \code{ngen} number of generations in each chain; \code{proposal} variance on the proposal distribution for the MCMC; \code{lrtest} logical value indicating whether to conduct a likelihood-ratio test of the hypothesis that some correlations are zero (does not appear to work in the present version); and \code{cleanup} remove PHYLIP input/output files after the analysis is completed (defaults to \code{cleanup = TRUE}).
\end{Details}
%
\begin{Value}
This function returns a list containing the results from threshml.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2005) Using the quantitative genetic threshold model for inferences between and within species \emph{Philosophical Transactions of the Royal Society London B}, 360, 1427-1434.

Felsenstein, J. (2012) A comparative method for both discrete and continuous characters using the threshold model. \emph{American Naturalist}, 179, 145-156.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rcontrast}{Rcontrast}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rtreedist}{R interface for treedist}{Rtreedist}
\keyword{phylogenetics}{Rtreedist}
%
\begin{Description}\relax
This function is an R interface for treedist in the PHYLIP package (Felsenstein 2013). treedist can be used to compute the distance between trees by two different methods.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rtreedist(trees, method=c("branch.score","symmetric"), path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{trees}] an object of class \code{"multiPhylo"}. (Or, under rare circumstances, an object of class \code{"phylo"}. See below.)
\item[\code{method}] method to compute the distance between trees. \code{method="branch.score"} is the branch score method of Kuhner \& Felsenstein (1994). \code{method="symmetric"} is the symmetric distance or Robinson-Foulds distance (Bourque 1978; Robinson \& Foulds 1981).
\item[\code{path}] path to the directory containing the executable treedist. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to treedist. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{trees2} object of class \code{"multiPhylo"} or \code{"phylo"} - if two sets of trees are to be compared; \code{rooted} logical value indicating whether trees should be treated as rooted (defaults to \code{rooted = FALSE}); \code{distances} argument telling treedist which distances to compute - could be \code{"all"} (all pairwise in \code{trees}), \code{"all.1to2"} (all in \code{trees} by all in \code{trees2}), \code{"adjacent"} (adjacent species in \code{trees} only), and \code{"corresponding"} (all corresponding trees in \code{trees} and \code{trees2}); and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

More information about the treedist program in PHYLIP can be found here \url{http://evolution.genetics.washington.edu/phylip/doc/treedist.html}.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 1989, 2013) should first be installed. Instructions for installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns a matrix of pairwise distances for \code{distances = "all"} and \code{distances = "all.1to2"}, or a named vector for \code{distances = "adjacent"} and \code{distances = "corresponding"}.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Bourque, M. (1978) \emph{Arbres de Steiner et reseaux dont certains sommets sont a localisation variable}. Ph.D. Dissertation, Universite de Montreal, Montreal, Quebec.

Felsenstein, J. (1989) PHYLIP--Phylogeny Inference Package (Version 3.2). \emph{Cladistics}, 5, 164-166.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Kuhner, M.K., Felsenstein, J. (1994) A simulation comparison of phylogeny algorithms under equal and unequal evolutionary rates. \emph{Molecular Biology and Evolution}, 11, 459-468.

Robinson, D.F., Foulds, L.R. (1981) Comparison of phylogenetic trees. \emph{Mathematical Biosciences}, 53, 131-147.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
trees<-rmtree(n=10,N=10)
D<-Rtreedist(trees,method="symmetric")

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{setPath}{Set path to the folder containing PHYLIP executables for the current R session}{setPath}
\aliasA{clearPath}{setPath}{clearPath}
\keyword{utilities}{setPath}
%
\begin{Description}\relax
Sets the path to the folder containing PHYLIP executables for the current R session. Once \code{setPath} has been used, Rphylip functions will no longer search for the path to the PHYLIP executable, using \code{path} instead.

\code{clearPath} clears the path.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
setPath(path)
clearPath()
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{path}] path to the folder contains PHYLIP executables.
\end{ldescription}
\end{Arguments}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
\inputencoding{utf8}
\HeaderA{setupOSX}{Help set up PHYLIP in Mac OS X}{setupOSX}
\keyword{phylogenetics}{setupOSX}
\keyword{utilities}{setupOSX}
%
\begin{Description}\relax
This function attempts to help set up PHYLIP on a Mac OS X machine.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
setupOSX(path=NULL)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{path}] path to the folder containing the PHYLIP package. If \code{path = NULL}, the R will search several commonly used directories.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
This function can be used to help set up PHYLIP (\url{http://evolution.genetics.washington.edu/phylip.html}) following the special instructions found here: \url{http://evolution.genetics.washington.edu/phylip/install.html}. \code{setupOSX} should only be run once - when PHYLIP is first installed.
\end{Details}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
setupOSX()

## End(Not run)
\end{ExampleCode}
\end{Examples}
\printindex{}
\end{document}
