\nonstopmode{}
\documentclass[a4paper]{book}
\usepackage[times,inconsolata,hyper]{Rd}
\usepackage{makeidx}
\usepackage[utf8,latin1]{inputenc}
% \usepackage{graphicx} % @USE GRAPHICX@
\makeindex{}
\begin{document}
\chapter*{}
\begin{center}
{\textbf{\huge Package `Rphylip'}}
\par\bigskip{\large \today}
\end{center}
\begin{description}
\raggedright{}
\item[Version]\AsIs{0.1-08}
\item[Date]\AsIs{2013-12-9}
\item[Title]\AsIs{Rphylip: An R interface for PHYLIP}
\item[Author]\AsIs{Liam J. Revell}
\item[Maintainer]\AsIs{Liam J. Revell }\email{liam.revell@umb.edu}\AsIs{}
\item[Depends]\AsIs{R (>= 2.10), ape (>= 3.0-10)}
\item[ZipData]\AsIs{no}
\item[Description]\AsIs{Rphylip provides an R interface for the PHYLIP package. All users of
Rphylip will thus first have to install the PHYLIP phylogeny methods program
package (Felsenstein 2013).
See http://evolution.genetics.washington.edu/phylip.html for more information
about installing PHYLIP.}
\item[License]\AsIs{GPL (>= 2)}
\item[URL]\AsIs{}\url{http://www.phytools.org/Rphylip}\AsIs{}
\item[Repository]\AsIs{}
\item[Date/Publication]\AsIs{2013-12-9 12:00:00 EDT}
\end{description}
\Rdcontents{\R{} topics documented:}
\inputencoding{utf8}
\HeaderA{Rphylip-package}{Rphylip: An R interface for PHYLIP}{Rphylip.Rdash.package}
\aliasA{Rphylip}{Rphylip-package}{Rphylip}
\keyword{package}{Rphylip-package}
%
\begin{Description}\relax
\pkg{Rphylip} provides an R interface for programs in the PHYLIP phylogeny methods package (Felsenstein 2013).
\end{Description}
%
\begin{Details}\relax
The complete list of functions can be displayed with \code{library(help = Rphylip)}.

Obviously, before any of the functions of this package can be used, users must first install PHYLIP (Felsenstein 2013). More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.

More information on \pkg{Rphylip} can be found at \url{http://www.phytools.org/Rphylip/} or \url{http://blog.phytools.org}.
\end{Details}
%
\begin{Author}\relax
Liam J. Revell

Maintainer: Liam J. Revell <liam.revell@umb.edu>
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.

Revell, L. J. (2013) Rphylip: An R interface for PHYLIP. R package version x-y.z.
\end{References}
\inputencoding{utf8}
\HeaderA{as.proseq}{Converts objects to protein sequences}{as.proseq}
\keyword{phylogenetics}{as.proseq}
\keyword{inference}{as.proseq}
\keyword{maximum likelihood}{as.proseq}
%
\begin{Description}\relax
Converts objects to class \code{"proseq"}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
as.proseq(x, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{x}] an object containing amino sequences. (Presently only objects of class \code{"phyDat"} are permitted.)
\item[\code{...}] optional arguments.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
An object of class \code{"proseq"} containing protein sequences.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{SeeAlso}\relax
\code{\LinkA{print.proseq}{print.proseq}}, \code{\LinkA{Rproml}{Rproml}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{opt.Rdnaml}{Parameter optimizer for Rdnaml}{opt.Rdnaml}
\keyword{phylogenetics}{opt.Rdnaml}
\keyword{inference}{opt.Rdnaml}
\keyword{maximum likelihood}{opt.Rdnaml}
%
\begin{Description}\relax
This function is an wrapper for \code{\LinkA{Rdnaml}{Rdnaml}} that attempts to optimize \code{gamma} (the alpha shape parameter of the gamma model of rate heterogeneity among sites), \code{kappa} (the transition:transversion ratio), and \code{bf} (the base frequencies).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
opt.Rdnaml(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing dnaml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{tree} fixed tree to use in optimization - if not provided, it will be estimated using \code{\LinkA{Rdnaml}{Rdnaml}} under the default conditions; \code{bounds} a list with bounds for optimization - for \code{kappa} and \code{gamma} this should be a two-element vector, whereas for \code{bf} this should be a 4 x 2 matrix with lower bounds in column 1 and upper bounds in column 2.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns a list with the following components: \code{kappa}, \code{gamma}, \code{bf} (see Details), and \code{logLik} (the log-likelihood of the fitted model).
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{opt.Rdnaml}{opt.Rdnaml}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
fit<-opt.Rdnaml(primates,bounds=list(kappa=c(0.1,40))
tree<-Rdnaml(primates,kappa=fit$kappa,gamma=fit$gamma,bf=fit$bf)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{primates}{Example datasets}{primates}
\aliasA{chloroplast}{primates}{chloroplast}
\keyword{datasets}{primates}
%
\begin{Description}\relax
\code{primates} is an object of class \code{"DNAbin"} containing nucleotide sequence data of mysterious origin for 12 species of primates. \code{chloroplast} is an object of class\code{"proseq"} containing a chloroplast alignment from the phangorn package (Schliep 2011).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
data(primates)
data(chloroplast)
\end{verbatim}
\end{Usage}
%
\begin{Format}
The data are stored as an object of class \code{"DNAbin"} or \code{"proseq"}.
\end{Format}
%
\begin{Source}\relax
Unknown.
\end{Source}
\inputencoding{utf8}
\HeaderA{print.proseq}{Print method protein sequences}{print.proseq}
\keyword{phylogenetics}{print.proseq}
\keyword{inference}{print.proseq}
\keyword{maximum likelihood}{print.proseq}
%
\begin{Description}\relax
Print method for an object of class \code{"proseq"}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
## S3 method for class 'proseq'
print(x, printlen=6, digits=3, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{x}] an object of class \code{"proseq"}.
\item[\code{printlen}] number of sequence names to print.
\item[\code{digits}] number of digits to print.
\item[\code{...}] optional arguments.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
Prints to screen.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{SeeAlso}\relax
\code{\LinkA{as.proseq}{as.proseq}}, \code{\LinkA{Rproml}{Rproml}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rconsense}{R interface for consense}{Rconsense}
\keyword{phylogenetics}{Rconsense}
%
\begin{Description}\relax
This function is an R interface for consense in the PHYLIP package (Felsenstein 2013). consense can be used to compute the consensus tree from a set of phylogenies. 
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rconsense(trees, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{trees}] an object of class \code{"multiPhylo"}.
\item[\code{path}] path to the directory containing the executable consense. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to consense. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{method} which can be \code{"extended"} (extended majority rule consensus, the default), \code{"strict"} (strict consensus), or regular majority rule consensus (\code{"majority"}); \code{"outgroup"} single taxon label or vector of taxa that should be used to root all trees before analysis; \code{rooted} logical value indicated whether to treat the input trees as rooted (defaults to \code{rooted = FALSE}); and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"}. For methods other than \code{method = "strict"}, \code{tree\$node.label} contains the proportion of phylogenies in \code{trees} containing that subtree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
trees<-rmtree(n=10,N=10)
tree<-Rconsense(trees)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rcontml}{R interface for contml}{Rcontml}
\keyword{phylogenetics}{Rcontml}
\keyword{inference}{Rcontml}
\keyword{maximum likelihood}{Rcontml}
%
\begin{Description}\relax
This function is an R interface for contml in the PHYLIP package (Felsenstein 2013). contml can be used for ML phylogeny estimation from gene frequency data or continuous characters. The continuous characters should be rotated so as to be uncorrelated before analysis.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rcontml(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] either (a) a \emph{matrix} of continuous valued traits (in columns) with rownames containing species names; or (b) a list of matrices in which each row contains the relative frequency of alleles at a locus for a species. In the latter case the rownames of each matrix in the list should contain the species names.
\item[\code{path}] path to the executable containing contml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to contml. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the model will be optimized on a fixed input topology; \code{global} perform global search (defaults to \code{global = TRUE}); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input/output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnaml}{Rdnaml}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rcontrast}{R interface for contrast}{Rcontrast}
\keyword{phylogenetics}{Rcontrast}
\keyword{comparative method}{Rcontrast}
%
\begin{Description}\relax
This function is an R interface for contrast in the PHYLIP package (Felsenstein 2013). contrast can be used to perform the among species phylogenetically independent contrasts method of Felsenstein (1985) and the within \& among species method of Felsenstein (2008).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rcontrast(tree, X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{tree}] object of class \code{"phylo"}.
\item[\code{X}] a \emph{matrix} of continuous valued traits (in columns) with rownames containing species names. For within-species contrasts analysis, the matrix should contain repeating (identical) row names for conspecifics.
\item[\code{path}] path to the executable containing contrast. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to contrast. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); and \code{cleanup} remove PHYLIP input/output files after the analysis is completed (defaults to \code{cleanup = TRUE}).
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (1985) Phylogenies and the comparative method. American Naturalist, 125, 1-15.

Felsenstein, J. (2008) Comparative methods with sampling error and within-species variation: Contrasts revisited and revised. American Naturalist, 171, 713-725.

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{pic}{pic}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rdnadist}{R interfaces for dnadist}{Rdnadist}
\keyword{phylogenetics}{Rdnadist}
\keyword{inference}{Rdnadist}
\keyword{maximum likelihood}{Rdnadist}
\keyword{distance method}{Rdnadist}
%
\begin{Description}\relax
This function is an R interface for dnadist in the PHYLIP package (Felsenstein 2013). dnadist can be used to estimate the evolutionary distances between DNA sequences under various models.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnadist(X, method=c("F84","K80","JC","LogDet"), path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{method}] method for calculating the distances. Can be \code{"F84"} (Kishino \& Hasegawa 1989; Felsenstein \& Churchill 1996), \code{"K80"} (Kimura 1980), \code{"JC"} (Jukes \& Cantor 1969), or \code{"LogDet"} (Barry \& Hartigan 1987; Lake 1994; Steel 1994; Lockhart et. al. 1994). Also \code{method="similarity"} computes the sequence similarity among the rows of \code{X}.
\item[\code{path}] path to the executable containing dnadist. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnadist. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{gamma} alpha shape parameter of a gamma model of rate heterogeneity among sites (defaults to no gamma rate heterogeneity); \code{kappa} transition:transversion ratio (defaults to \code{kappa = 2.0}); \code{rates} vector of rates (defaults to single rate); \code{rate.categories} vector of rate categories corresponding to the order of \code{rates}; \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{bf} vector of base frequencies in alphabetical order (i.e., A, C, G, \& T) - if not provided, then defaults to empirical frequencies; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"dist"}.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rneighbor}{Rneighbor}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
D<-Rdnadist(primates,kappa=10)
tree<-Rneighbor(D)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rdnaml}{R interfaces for dnaml and dnamlk}{Rdnaml}
\aliasA{Rdnamlk}{Rdnaml}{Rdnamlk}
\keyword{phylogenetics}{Rdnaml}
\keyword{inference}{Rdnaml}
\keyword{maximum likelihood}{Rdnaml}
%
\begin{Description}\relax
This function is an R interface for dnaml in the PHYLIP package (Felsenstein 2013). dnaml can be used for ML phylogeny estimation from DNA sequences.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnaml(X, path=NULL, ...)
Rdnamlk(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing dnaml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnaml or dnamlk. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the model will be optimized on a fixed input topology; \code{kappa} transition:transversion ratio (defaults to \code{kappa = 2.0}); \code{bf} vector of base frequencies in alphabetical order (i.e., A, C, G, \& T) - if not provided, then defaults to empirical frequencies; \code{rates} vector of rates (defaults to single rate); \code{rate.categories} vector of rate categories corresponding to the order of \code{rates}; \code{gamma} alpha shape parameter of a gamma model of rate heterogeneity among sites (defaults to no gamma rate heterogeneity); \code{ncat} number of rate categories for the gamma model; \code{inv} proportion of invariant sites for the invariant sites model (defaults to \code{inv = 0}); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{speedier} speedier but rougher analysis (defaults to \code{speedier = FALSE}); \code{global} perform global search (defaults to \code{global = TRUE}); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Finally \code{clock=TRUE} enforces a molecular clock. The argument \code{clock} is only available for \code{Rdnaml}. If \code{clock=TRUE} then dnamlk is used internally. For \code{Rdnamlk} a molecular clock is assumed, thus \code{Rdnaml(...,clock=TRUE)} and \code{Rdnamlk(...)} are equivalent.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{opt.Rdnaml}{opt.Rdnaml}}, \code{\LinkA{Rcontml}{Rcontml}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
tree<-Rdnaml(primates,kappa=10)
clockTree<-Rdnamlk(primates,kappa=10)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rdnapars}{R interface for dnapars}{Rdnapars}
\keyword{phylogenetics}{Rdnapars}
\keyword{inference}{Rdnapars}
\keyword{parsimony}{Rdnapars}
%
\begin{Description}\relax
This function is an R interface for dnapars in the PHYLIP package (Felsenstein 2013). dnapars can be used for MP phylogeny estimation from DNA sequences.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnapars(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing dnapars. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnapars. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the parsimony score will be computed on a fixed input topology; \code{thorough} logical value indicating whether to conduct a more thorough search (defaults to \code{thorough=TRUE}); \code{nsave} number of trees to save if multiple equally parsimonious trees are found (defaults to \code{nsave=10000}); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{threshold} threshold value for threshold parsimony (defaults to ordinary parsimony); \code{transversion} logical value indicating whether to use transversion parsimony (defaults to \code{transversion=FALSE}); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnaml}{Rdnaml}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
tree<-Rdnapars(primates)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rdnapenny}{R interface for dnapenny}{Rdnapenny}
\keyword{phylogenetics}{Rdnapenny}
\keyword{inference}{Rdnapenny}
\keyword{parsimony}{Rdnapenny}
%
\begin{Description}\relax
This function is an R interface for dnapenny in the PHYLIP package (Felsenstein 2013). dnapenny performs branch \& bound parsimony searching following Hendy \& Penny (1982).
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rdnapenny(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing dnapenny. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to dnapenny. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{groups} number of groups of 1,000 trees (defaults to \code{groups = 10000}); \code{report} reporting frequency, in numbers of trees (defaults to \code{report = 1000}); \code{simple} simple branch \& bound (defaults to \code{simple = TRUE}); \code{threshold} threshold value for threshold parsimony (defaults to ordinary parsimony); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} or \code{"multiPhylo"} that is the tree or trees with the best parsimony score. \code{tree\$score} gives the parsimony score, for \code{"phylo"} object \code{tree}.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnaml}{Rdnaml}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
tree<-Rdnapenny(primates)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{read.protein}{Reads protein sequences from file in multiple formats}{read.protein}
\keyword{phylogenetics}{read.protein}
\keyword{inference}{read.protein}
\keyword{maximum likelihood}{read.protein}
%
\begin{Description}\relax
Reads protein sequences from a file.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
read.protein(file, format="fasta", ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{file}] file name for file containing protein sequences.
\item[\code{format}] format of input file. Permitted formats are \code{"fasta"} and \code{"sequential"}. See \code{\LinkA{read.dna}{read.dna}} for more information.
\item[\code{...}] optional arguments.
\end{ldescription}
\end{Arguments}
%
\begin{Value}
An object of class \code{"proseq"} containing protein sequences.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{SeeAlso}\relax
\code{\LinkA{as.proseq}{as.proseq}}, \code{\LinkA{print.proseq}{print.proseq}}, \code{\LinkA{Rproml}{Rproml}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rneighbor}{R interface for neighbor}{Rneighbor}
\keyword{phylogenetics}{Rneighbor}
\keyword{inference}{Rneighbor}
\keyword{distance method}{Rneighbor}
%
\begin{Description}\relax
This function is an R interface for neighbor in the PHYLIP package (Felsenstein 2013). neighbor can be used for neighbor-joining and UPGMA phylogeny inference.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rneighbor(D, path=NULL , ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{D}] a distance matrix as an object of class \code{"matrix"} or \code{"dist"}. If a matrix, then \code{D} should be symmetrical with a diagonal of zeros.
\item[\code{path}] path to the executable containing neighbor. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to neighbor. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{method} - can be \code{"NJ"} or \code{"nj"} (for neighbor-joining), or \code{"UPGMA"} or \code{"UPGMA"} (for UPGMA); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired (only works for \code{method = "NJ"}, UPGMA trees are already rooted); and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the NJ or UPGMA tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnaml}{Rdnaml}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(primates)
D<-dist.dna(data(primates),model="JC")
tree<-Rneighbor(D)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rproml}{R interfaces for proml and promlk}{Rproml}
\aliasA{Rpromlk}{Rproml}{Rpromlk}
\keyword{phylogenetics}{Rproml}
\keyword{inference}{Rproml}
\keyword{maximum likelihood}{Rproml}
%
\begin{Description}\relax
This function is an R interface for proml in the PHYLIP package (Felsenstein 2013). proml can be used for ML phylogeny estimation from amino acid sequences.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rproml(X, path=NULL, ...)
Rpromlk(X, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{X}] an object of class \code{"DNAbin"}.
\item[\code{path}] path to the executable containing proml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to proml or promlk. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{tree} object of class \code{"phylo"} - if supplied, then the model will be optimized on a fixed input topology; \code{model} amino acid model - could be \code{"JTT"} (Jones et al. 1992), \code{"PMB"} (Veerassamy et al. 2003), or \code{"PAM"} (Dayhoff \& Eck 1968; Dayhoff et al. 1979; Koisol \& Goldman 2005); \code{rates} vector of rates (defaults to single rate); \code{rate.categories} vector of rate categories corresponding to the order of \code{rates}; \code{gamma} alpha shape parameter of a gamma model of rate heterogeneity among sites (defaults to no gamma rate heterogeneity); \code{ncat} number of rate categories for the gamma model; \code{inv} proportion of invariant sites for the invariant sites model (defaults to \code{inv = 0}); \code{weights} vector of weights of length equal to the number of columns in \code{X} (defaults to unweighted); \code{speedier} speedier but rougher analysis (defaults to \code{speedier = FALSE}); \code{global} perform global search (defaults to \code{global = TRUE}); \code{random.order} add taxa to tree in random order (defaults to \code{random.order = TRUE}); \code{random.addition} number of random addition replicates for \code{random.order = TRUE} (defaults to \code{random.addition = 10}); \code{outgroup} outgroup if outgroup rooting of the estimated tree is desired; and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Finally \code{clock=TRUE} enforces a molecular clock. The argument \code{clock} is only available for \code{Rproml}. If \code{clock=TRUE} then promlk is used internally. For \code{Rpromlk} a molecular clock is assumed, thus \code{Rproml(...,clock=TRUE)} and \code{Rpromlk(...)} are equivalent. Note that in PHYLIP 3.695 my tests of promlk yielded peculiar results (all branch lengths zero length, random topology), so I'm not sure what to make of that.

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns an object of class \code{"phylo"} that is the optimized tree.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rdnaml}{Rdnaml}}, \code{\LinkA{Rcontml}{Rcontml}}
\end{SeeAlso}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
data(chloroplast)
tree<-Rproml(chloroplast)

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{Rthreshml}{R interface for threshml}{Rthreshml}
\keyword{phylogenetics}{Rthreshml}
\keyword{comparative method}{Rthreshml}
%
\begin{Description}\relax
This function is an R interface for threshml in the PHYLIP package (Felsenstein 2013). threshml fits the threshold model of Felsenstein (2005; 2012).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rthreshml(tree, X, types=NULL, path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{tree}] object of class \code{"phylo"}.
\item[\code{X}] a \emph{matrix} of continuous valued traits (in columns) with rownames containing species names. For within-species contrasts analysis, the matrix should contain repeating (identical) row names for conspecifics.
\item[\code{types}] character vector containing the types (e.g., \code{"discrete"}, \code{"continuous"}). If \code{types} are not supplied, \code{Rthreshml} will try to figure them out.
\item[\code{path}] path to the executable containing contrast. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to threshml. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{burnin} burnin generations for the MCMC; \code{nchain} number of chains of the MCMC; \code{ngen} number of generations in each chain; \code{proposal} variance on the proposal distribution for the MCMC; \code{lrtest} logical value indicating whether to conduct a likelihood-ratio test of the hypothesis that some correlations are zero (does not appear to work in the present version); and \code{cleanup} remove PHYLIP input/output files after the analysis is completed (defaults to \code{cleanup = TRUE}).
\end{Details}
%
\begin{Value}
This function returns a list containing the results from threshml.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax

Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{SeeAlso}\relax
\code{\LinkA{Rcontrast}{Rcontrast}}
\end{SeeAlso}
\inputencoding{utf8}
\HeaderA{Rtreedist}{R interface for treedist}{Rtreedist}
\keyword{phylogenetics}{Rtreedist}
%
\begin{Description}\relax
This function is an R interface for treedist in the PHYLIP package (Felsenstein 2013). treedist can be used to compute the distance between trees by two different methods.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
Rtreedist(trees, method=c("branch.score","symmetric"), path=NULL, ...)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{trees}] an object of class \code{"multiPhylo"}. (Or, under rare circumstances, an object of class \code{"phylo"}. See below.)
\item[\code{method}] method to compute the distance between trees. \code{method="branch.score"} is the branch score method of Kuhner \& Felsenstein (1994). \code{method="symmetric"} is the symmetric distance or Robinson-Foulds distance (Bourque 1978; Robinson \& Foulds 1981).
\item[\code{path}] path to the directory containing the executable treedist. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\item[\code{...}] optional arguments to be passed to treedist. See details for more information.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
Optional arguments include the following: \code{quiet} suppress some output to R console (defaults to \code{quiet = FALSE}); \code{trees2} object of class \code{"multiPhylo"} or \code{"phylo"} - if two sets of trees are to be compared; \code{rooted} logical value indicating whether trees should be treated as rooted (defaults to \code{rooted = FALSE}); \code{distances} argument telling treedist which distances to compute - could be \code{"all"} (all pairwise in \code{trees}), \code{"all.1to2"} (all in \code{trees} by all in \code{trees2}), \code{"adjacent"} (adjacent species in \code{trees} only), and \code{"corresponding"} (all corresponding trees in \code{trees} and \code{trees2}); and \code{cleanup} remove PHYLIP input \& output files after the analysis is completed (defaults to \code{cleanup = TRUE}).

Obviously, use of any of the functions of this package requires that PHYLIP (Felsenstein 2013) should first be installed. More information about installing PHYLIP can be found on the PHYLIP webpage: \url{http://evolution.genetics.washington.edu/phylip.html}.
\end{Details}
%
\begin{Value}
This function returns a matrix of pairwise distances for \code{distances = "all"} and \code{distances = "all.1to2"}, or a named vector for \code{distances = "adjacent"} and \code{distances = "corresponding"}.
\end{Value}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
trees<-rmtree(n=10,N=10)
D<-Rtreedist(trees,method="symmetric")

## End(Not run)
\end{ExampleCode}
\end{Examples}
\inputencoding{utf8}
\HeaderA{setupOSX}{Help set up PHYLIP in Mac OS X}{setupOSX}
\keyword{phylogenetics}{setupOSX}
\keyword{utilities}{setupOSX}
%
\begin{Description}\relax
This function attempts to help set up PHYLIP on a Mac OS X machine.
\end{Description}
%
\begin{Usage}
\begin{verbatim}
setupOSX(path=NULL)
\end{verbatim}
\end{Usage}
%
\begin{Arguments}
\begin{ldescription}
\item[\code{path}] path to the executable containing dnaml. If \code{path = NULL}, the R will search several commonly used directories for the correct executable file.
\end{ldescription}
\end{Arguments}
%
\begin{Details}\relax
This function can be used to help set up PHYLIP (\url{http://evolution.genetics.washington.edu/phylip.html}) following the special instructions found here: \url{http://evolution.genetics.washington.edu/phylip/install.html}. \code{setupOSX} should only be run once - when PHYLIP is first installed.
\end{Details}
%
\begin{Author}\relax
Liam Revell \email{liam.revell@umb.edu}
\end{Author}
%
\begin{References}\relax
Felsenstein, J. (2013) PHYLIP (Phylogeny Inference Package) version 3.695. Distributed by the author. Department of Genome Sciences, University of Washington, Seattle.
\end{References}
%
\begin{Examples}
\begin{ExampleCode}
## Not run: 
setupOSX()

## End(Not run)
\end{ExampleCode}
\end{Examples}
\printindex{}
\end{document}
